### PR TITLE
Adapt frontend/backend for create_account request

### PR DESCRIPTION
**Pull request by zaucker**

### CODE REVIEW

File: t/webservice.t [hunk 1]
```diff
@@ -81,15 +81,12 @@ transactionally {
...
-            :organisation('Org'),
-        );
-        ok $newUser = $ws.create-account($user, $userData), "Create new account";
-        is $newUser.username, 'foo@bar.ch', "User has expected username";
+        ok $username = $ws.create-account(
+            $user,
+            'foo@bar.com', 'myPassword', 'myKey',
+            'Erika', 'Mustermann', 'MyOrg'
+        ), "Create new account";
+        is $username, 'foo@bar.com', "User has expected username";
```
Comment (jnthn) [inline t/webservice.t:89]: Also add a test for the case where `Any` is passed for the things that are optional, to cover that case.

Comment (zaucker) [reply t/webservice.t]: Added another test, some error handling, and OpenAPI validation.

File: lib/Agrammon/DB/User.pm6 [hunk 2]
```diff
@@ -15,13 +15,21 @@ class X::Agrammon::DB::User::NoUsername is Exception {
     }
 }
 
+class X::Agrammon::DB::User::CreateFailed is Exception {
+    has Str $.user-name is required;
+
+    method message {
+        "Couldn't create user $.username";
```
Comment (jnthn) [inline lib/Agrammon/DB/User.pm6:22]: Should be `$.user-name` (but you don't really need it to be a virtual call at all, so you can access the underlying attribute directly as `$!user-name`, and those typos are caught at compile time).

File: lib/Agrammon/Web/Routes.pm6 [hunk 3]
```diff
@@ -66,6 +66,21 @@ sub static-content($root) {
...
+        # working
+        operation 'createAccount', -> LoggedIn $user {
+            request-body -> (:$email!, :$password!, :$key, :$firstname, :$lastname, :$org) {
+                my $username = $ws.create-account($user, $email, $password, $key, $firstname, $lastname, $org);
+                content 'application/json', { :$username };
+                CATCH {
+                    note "$_";
+                    when X::Agrammon::DB::User::AlreadyExists | X::Agrammon::DB::User::CreateFailed {
+                        conflict 'application/json', %(
+                            error => .message
```
Comment (jnthn) [inline lib/Agrammon/Web/Routes.pm6:78]: That the bug above isn't caught means we don't have a test covering this line.

File: share/agrammon.openapi [hunk 4]
```diff
@@ -3,6 +3,56 @@ info:
...
+                                - password
+                            properties:
+                                email:
+                                    description: User's email used as username
+                                    type: string
+                                password:
+                                    description: Inital password
+                                    type: string
+                                key:
+                                    description: Validation keu
```
Comment (jnthn) [inline share/agrammon.openapi:27]: keu?

**Review by jnthn — CHANGES_REQUESTED** [review]

File: lib/Agrammon/DB/User.pm6 (hunk 2 above)
Comment (zaucker) [reply lib/Agrammon/DB/User.pm6:22]: Fixed.

File: lib/Agrammon/Web/Routes.pm6 (hunk 3 above)
Comment (zaucker) [reply lib/Agrammon/Web/Routes.pm6:78]: Added test.

File: share/agrammon.openapi (hunk 4 above)
Comment (zaucker) [reply share/agrammon.openapi:27]: key ... fixed

File: lib/Agrammon/Web/Routes.pm6 (hunk 3 above)
Comment (zaucker) [reply lib/Agrammon/Web/Routes.pm6:78]: With the current setup I don't see that `X::Agrammon::DB::User::CreateFailed` could be thrown without a DB-failure.
I left it in there anyway and added a test for it in `user.t`.

**Comment by zaucker** [comment]
The whole thing got a bit more complex, as I exposed the role attribute as well (not yet used in the frontend, thus optional with default). But now the API is complete

File: lib/Agrammon/DB/User.pm6 [hunk 5]
```diff
@@ -30,26 +46,34 @@ class Agrammon::DB::User does Agrammon::DB {
         $!username = $username;
     }
 
-    method create-account(Str $role-name) {
-        warn "*** create account";
-        die X::Agrammon::DB::User::Exists.new(:username($!username)) if self.exists;
+    method create-account($role-name) {
+
+        my $role = $role-name ?? $role-name !! 'user';
```
Comment (jnthn) [inline lib/Agrammon/DB/User.pm6:51]: Or just `$role-name || 'user'`

File: lib/Agrammon/Web/Routes.pm6 (hunk 3 above)
Comment (jnthn) [reply lib/Agrammon/Web/Routes.pm6:79]: Whether all of these should really be 409 is another question, though... I mean, already exists is obviously a conflict. But "unknown role" doesn't really feel like one. It's good we're using a HTTP error code, but perhaps we should try and pick appropriate ones.

File: share/agrammon.openapi [hunk 6]
```diff
@@ -3,6 +3,62 @@ info:
...
+                        application/json:
+                            schema:
+                                type: object
+                                required:
+                                    - username
+                                properties:
+                                    username:
+                                        type: string
+                '409':
+                    description: User already exists
```
Comment (jnthn) [inline share/agrammon.openapi:57]: And this only documents the conflict case too

File: lib/Agrammon/DB/User.pm6 (hunk 5 above)
Comment (zaucker) [reply lib/Agrammon/DB/User.pm6:51]: Done

File: share/agrammon.openapi (hunk 6 above)
Comment (zaucker) [reply share/agrammon.openapi:69]: I added a few more status codes.

**Review by jnthn — APPROVED** [review]

**Pull request merged**